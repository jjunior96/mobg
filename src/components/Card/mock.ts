import { v4 } from 'uuid';

import { CardProps } from '.';

export const CardMock: CardProps[] = [
  {
    id: v4(),
    type: 'house',
    location: 'Avenida Brasil',
    price: '1.160',
    meters: 64,
    rooms: 2,
    garage: 2,
    img: '/img/house-1.jpeg'
  },
  {
    id: v4(),
    type: 'apartment',
    location: 'Avenida Uruguai',
    price: '1.000',
    meters: 58,
    rooms: 2,
    garage: 1,
    img: '/img/house-1.jpeg'
  },
  {
    id: v4(),
    type: 'house',
    location: 'Avenida Brasil',
    price: '2.310',
    meters: 140,
    rooms: 4,
    garage: 4,
    img: '/img/house-1.jpeg'
  }
];
