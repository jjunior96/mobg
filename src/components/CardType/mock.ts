import { v4 } from 'uuid';

import { CardTypeProps } from '.';

export const CardTypeMock: CardTypeProps[] = [
  {
    id: v4(),
    type: 'Apartamentos',
    description: 'Lofts, studios, kitnets...',
    amount: 126,
    link: '#',
    img: '/img/house-1.jpeg'
  }
];
