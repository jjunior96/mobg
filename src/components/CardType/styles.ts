import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 80rem;

    border-radius: ${theme.border.radius};

    overflow: hidden;
  `}
`;

export const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  height: 28rem;
`;

export const InfoContainer = styled.div`
  ${({ theme }) => css`
    width: 50%;

    height: 100%;

    padding: ${theme.spacings.small};

    display: flex;
    flex-direction: column;
    justify-content: space-between;

    border: 0.2rem solid ${theme.colors.gray_100};
    border-right: none;
  `}
`;

export const InfoContent = styled.div`
  display: flex;
  flex-direction: column;
`;

export const InfoDescription = styled.p`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.small};

    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.blue_200};
  `}
`;

export const InfoAmount = styled.p`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xxsmall};

    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.blue_200};
  `}
`;

export const ImgContainer = styled.img`
  width: 50%;

  height: 100%;

  object-fit: cover;
`;

export const SeeMoreLink = styled.a`
  ${({ theme }) => css`
    text-decoration: none;

    line-height: 1;

    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.blue_200};

    width: fit-content;

    display: flex;
    align-items: center;

    width: fit-content;

    svg {
      margin-left: ${theme.spacings.xxsmall};

      color: ${theme.colors.blue_200};

      &:hover {
        color: ${theme.colors.primary};
      }
    }

    &:hover {
      color: ${theme.colors.primary};
    }
  `}
`;
