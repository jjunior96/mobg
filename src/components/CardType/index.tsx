import { MdArrowForwardIos as ArrowLeftIcon } from 'react-icons/md';

import Heading from 'components/Heading';

import * as S from './styles';

export type CardTypeProps = {
  id: string;
  type: 'Apartamentos' | 'Casa';
  description: string;
  amount: number;
  link: string;
  img: string;
};

const CardType = ({ type, description, amount, link, img }: CardTypeProps) => {
  return (
    <S.Container>
      <S.Content>
        <S.InfoContainer>
          <S.InfoContent>
            <Heading size="large">{type}</Heading>

            <S.InfoDescription>{description}</S.InfoDescription>
            <S.InfoAmount>{amount} unidades</S.InfoAmount>
          </S.InfoContent>

          <S.SeeMoreLink href={link}>
            <p>Ver mais</p>
            <ArrowLeftIcon size="16" />
          </S.SeeMoreLink>
        </S.InfoContainer>
        <S.ImgContainer src={img} />
      </S.Content>
    </S.Container>
  );
};

export default CardType;
