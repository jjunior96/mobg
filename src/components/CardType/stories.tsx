import { Story, Meta } from '@storybook/react/types-6-0';

import CardType from '.';

export default {
  title: 'CardType',
  component: CardType
} as Meta;

export const Default: Story = () => <CardType />;
