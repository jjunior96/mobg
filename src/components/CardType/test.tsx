import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import CardType from '.';
import { CardTypeMock } from './mock';

const props = CardTypeMock[0];

describe('<CardType />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<CardType {...props} />);

    expect(screen.getByText(/apartamentos/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
