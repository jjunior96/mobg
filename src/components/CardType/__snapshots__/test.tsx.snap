// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardType /> should render the heading 1`] = `
.c4 {
  font-size: 3.2rem;
}

.c0 {
  width: 100%;
  max-width: 80rem;
  border-radius: 0.8rem;
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 28rem;
}

.c2 {
  width: 50%;
  height: 100%;
  padding: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: 0.2rem solid #e6e6eb;
  border-right: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  margin-top: 2.4rem;
  font-size: 1.6rem;
  color: #5d5e7a;
}

.c6 {
  margin-top: 1rem;
  font-size: 1.6rem;
  color: #5d5e7a;
}

.c8 {
  width: 50%;
  height: 100%;
  object-fit: cover;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: 1;
  font-size: 1.6rem;
  color: #5d5e7a;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c7 svg {
  margin-left: 1rem;
  color: #5d5e7a;
}

.c7 svg:hover {
  color: #5c63ef;
}

.c7:hover {
  color: #5c63ef;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Apartamentos
        </h2>
        <p
          class="c5"
        >
          Lofts, studios, kitnets...
        </p>
        <p
          class="c6"
        >
          126
           unidades
        </p>
      </div>
      <a
        class="c7"
        href="#"
      >
        <p>
          Ver mais
        </p>
        <svg
          fill="currentColor"
          height="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </a>
    </div>
    <img
      class="c8"
      src="/img/house-1.jpeg"
    />
  </div>
</div>
`;
